### DISCUSSION TITLE
How to use `web fetch` to redirect？

### DISCUSSION THREAD

**Discussion by qianjiachun** [remix-run/remix#5859] [Q&A]
For example, here's an action to log out:
```
// app/routes/api/user/logout.tsx

export const action: ActionFunction = () => {
  return redirect("/", {
    headers: {
      "Set-Cookie": `token=;path=/;httpOnly;`
    }
  });
};
```

When I use `useFetcher` to submit, the page redirects successfully.

![image](https://user-images.githubusercontent.com/43806319/226542283-7c146e8e-f570-413d-b363-27717833411c.png)

This seems to be the code for its implementation:
> ```ts
> ```
> https://github.com/remix-run/remix/blob/ef26f7671a9619966a6cfa3c39e196d44fbf32cf/packages/remix-server-runtime/server.ts#L107-L116

But personally, I am more familiar with `web fetch`😂.(Because I didn't find a good solution to synchronously get the data after the useFetcher submit)
```
import fetch from "@remix-run/web-fetch";
fetch("/api/user/logout")
```

![image](https://user-images.githubusercontent.com/43806319/226542879-eca7dfb3-3f89-4be9-b1a1-ee7bb4ac4d14.png)

The status was 302, and it seems that need to handle it myself like `if (response.redirected) window.location.href = xxx` which cause the page reload.

🙏So I really want to know how to use `web fetch` in remix to redirect.

**Comment by sergiodxa** [comment]
The `@remix-run/web-fetch` package is more internal, that's why it's also not documented, if you want to follow a redirect you will have to do it yourself, what you can do is to use the `useNavigate` hook to get the `navigate` function and use that to move the user to the new URL.

**Reply by qianjiachun** [reply]
Thank you for your answer!
In fact, the request function I want is a pure function (return a Promise), without hooks. So this demand cannot be realized temporarily, right?😂
Last question. In addition to the `useFetcher ({onSuccess(){}, onError(){}})` found in the community, is there a better way to synchronously get the `submit` data, just like `async` `await`.
The usage scenario is similar to the Infinite scroll list (get the next page of data and append it to the end of the array).
